File: ws/carvajal/u2/ws11ClassDiagram/Farmsystem/src/main/java/ec/edu/espe/farmsymulator/model/Pig.java
package ec.edu.espe.farmsymulator.model;

import java.util.Date;

/**
 *
 * @author Josue Carvajal, THE ART OF PROGRAMMING, @ESPE
 */
public class Pig extends FarmAnimal {

    @Override
    public String toString() {
        return "Pig{" + super.toString()+'}';
    }

    
    
    public Pig(int id, String breed, Date bornOn, String gender, boolean isAbleToReproduce, float weight, Cage cage) {
        super(id, breed, bornOn, gender, isAbleToReproduce, weight, cage);
    }

    
}

File: hw/quina/u2/hw19Inheritance/FarmSystemPy/ec_edu_espe_model/Pig.py
"""
Author: Maryuri Quiña
Description: Farm System – Inheritance example in Python.
"""

from datetime import date
from .FarmAnimal import FarmAnimal
from .Cage import Cage


class Pig(FarmAnimal):
    def __init__(
        self,
        animal_id: int,
        breed: str,
        born_on: date,
        gender: str,
        is_able_to_reproduce: bool,
        weight: float,
        cage: Cage,
    ):
        super().__init__(animal_id, breed, born_on, gender, is_able_to_reproduce, weight, cage)

    def __str__(self) -> str:
        return f"Pig({super().__str__()})"
